### PR TITLE
wip: Introduce the in-repo-config for evolution repo

### LINKED ISSUE #840: Move `org.yaml` to evolution repo
This is the mirror issue of https://github.com/falcosecurity/evolution/issues/176.

**Motivation**

As the evolution repository is now the central point for the evolution of the project and its ecosystem, and as the falcosecurity organization is formed by the community, and as the point of contact of the contributions are in git (Github), is correct that the source of truth as a declarative configuration of the Github organization is put in the `evolution` repository.

**Feature**

- [ ] move [org.yaml](https://github.com/falcosecurity/test-infra/blob/master/config/org.yaml)
from `test-infra` to `evolution`. (https://github.com/falcosecurity/evolution/pull/274)
- [ ] update Peribolos job [repository](https://github.com/falcosecurity/test-infra/blob/master/config/jobs/peribolos/peribolos.yaml#L2) from _test-infra_ to _evolution_

**Alternatives**

**Additional context**
cc @falcosecurity/evolution-maintainers

### PR DESCRIPTION
**Pull request by maxgio92**
This PR introduces the [in_repo_config](https://github.com/kubernetes/test-infra/blob/master/prow/inrepoconfig.md#inrepoconfig), only for [evolution](https://github.com/falcosecurity/evolution) repo for now.

Moreover, it adds a presubmit prowjob config check, for PR opened in the [evolution](https://github.com/falcosecurity/evolution) repository. This is to alert evolution contributors on job configuration errors.

In detail, the jobs are expected in the evolution repo under the `/.prow/` directory.

Fixes #840

### CODE REVIEW

File: config/jobs/check-prow-config/check-prow-config.yaml [hunk 1]
```diff
@@ -1,22 +1,39 @@
...
+        command:
+        - checkconfig
+        args:
+        - --config-path=config/config.yaml
+        - --job-config-path=config/jobs
+        - --plugin-config=config/plugins.yaml
+      nodeSelector:
+        Archtype: "x86"
+  falcosecurity/evolution:
+  - name: check-prow-config-evolution
```
Comment (leogr) [inline config/jobs/check-prow-config/check-prow-config.yaml:21]: Should it run only when the `org.yaml` file has changed?

Comment (maxgio92) [reply config/jobs/check-prow-config/check-prow-config.yaml:21]: I agree @leogr. Good catch! Thanks

**Comment by zuc** [comment]
I think we should also provide support for autobump for in-repo job definitions.
cc @maxgio92 

/hold

**Comment by maxgio92** [comment]
Ok @zuc, let's find a solution to that. Thanks

File: config/jobs/check-prow-config/check-prow-config.yaml (hunk 1 above)
Comment (maxgio92) [reply config/jobs/check-prow-config/check-prow-config.yaml:21]: Applied

**Comment by maxgio92** [comment]
Hey @zuc please take a look at the last two commits (https://github.com/falcosecurity/test-infra/pull/841/commits/36768f80b2a06e45f2ff2c29fa7a5e7be4d33199 and https://github.com/falcosecurity/test-infra/pull/841/commits/ba53927cf76ebbf9d497f6eb707e76a7d915a738).

They aim to:
- allow to bump also job configurations only, as the case of the `evolution` repository (in detail: https://github.com/falcosecurity/evolution/pull/193)
- add the related autobump job for the `evolution` in-repo job configs.

**Comment by maxgio92** [comment]
As discussed with @leogr, @FedeDP, @cappellinsamuele, as the direction is to leverage as much as possible Github Actions for repo-specific jobs, I think it's not worth to introduce in-repo config as of now.

Instead, I'm going to open a PR dedicated only to #840, and one for refactoring of autobumper script.

/close

**Pull request closed without merging**